pkg/database/mysql.go

```diff
@@ -0,0 +1,26 @@
+package database
+
+
+type MySQLDatabase struct {
+	DB *bun.DB
+}
+
+var (
+	once sync.Once
+	dbInstance *MySQLDatabase
+)
+
+func NewMySQLDatabase(conf *entities.Config) *MySQLDatabase {
+	once.Do(func() {
+		sqlDB, err := sql.Open("mysql", fmt.Sprintf("%s:%s@tcp(%s:%s)/%s", conf.DBUser, conf.DBPassword, conf.DBHost, conf.DBPort, conf.DBName))
+		if err != nil {
+			panic(err)
+		}
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider returning error instead of panic</b></div>
<div id="fix">

Consider returning an error instead of using `panic()` when database connection fails. This would allow better error handling at the caller level.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">


```
 @@ -13,10 +13,10 @@
 func NewMySQLDatabase(conf *entities.Config) (*MySQLDatabase, error) {
 	var err error
 	once.Do(func() {
 		sqlDB, err := sql.Open("mysql", fmt.Sprintf("%s:%s@tcp(%s:%s)/%s", conf.DBUser, conf.DBPassword, conf.DBHost, conf.DBPort, conf.DBName))
 		if err != nil {
 -			panic(err)
 +			return nil, err
 		}
 		dbInstance = &MySQLDatabase{DB: bun.NewDB(sqlDB, mysqldialect.New())}
 	})
 -	return dbInstance
 +	return dbInstance, err
```

</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/aakashgusain1990/productivite/pull/14#issuecomment-2563464674>#148ccb</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged